runner/src/runner.rs

```diff
@@ -0,0 +1,330 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use alloc::collections::BTreeSet;
+use alloc::vec;
+use alloc::vec::Vec;
+
+use gear_backend_common::{funcs::EXIT_TRAP_STR, Environment};
+use gear_core::{
+    env::Ext as EnvExt,
+    gas::{ChargeResult, GasCounter},
+    memory::{MemoryContext, PageNumber},
+    message::{
+        IncomingMessage, Message, MessageContext, MessageId, MessageIdGenerator, MessageState,
+    },
+    program::{Program, ProgramId},
+};
+
+use crate::configs::{AllocationsConfig, BlockInfo, EntryPoint};
+use crate::ext::Ext;
+use crate::ids::BlakeMessageIdGenerator;
+
+const EXIT_CODE_PANIC: i32 = 1;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum ExecutionOutcome {
+    Done,
```

Comment:
where is `Wait` ?

Reply:
sticked in `Done` variant as a `wait` flag